rmc/utils/constraint.py

```diff
@@ -955,214 +933,235 @@ def get_rescue_transcripts_and_create_no_breaks_ht(overwrite: bool) -> None:
+    if len(single_search_round_nums) == 0:
+        raise DataException("No search rounds recorded, please double-check!")
+    if len(single_search_round_nums) == 1:
+        logger.warning(
+            "Only one break search round recorded. Either no breaks were found or break search is not complete, please double-check!"
         )
+    return single_search_round_nums
 
-    else:
-        logger.info(
-            "Getting info for last section of transcript (after largest breakpoint pos)..."
-        )
-        ht = ht.filter(ht.locus.position > ht.break_pos[-1])
-        ht = ht.annotate(
-            # Get last position from break_pos list and add 1 since it isn't included in the region
-            # Use the transcript end position as the end position since this is the last section
-            section_start_pos=ht.break_pos[-1] + 1,
-            section_end_pos=ht.end_pos,
-        )
 
-    ht = ht.annotate(section=hl.format("%s_%s", ht.transcript, str(section_num)))
-    ht = get_subsection_exprs(
-        ht, "transcript", "observed", "mu_snp", "total_mu", "total_exp"
+def merge_round_no_break_ht(is_rescue: bool, search_num: int) -> hl.Table:
+    """
+    Get merged single and simultaneous search no-break table from a given round of break search.
+
+    :param is_rescue: Whether to operate on search in rescue pathway.
+    :param search_num: Search iteration number
+        (e.g., second round of searching for single break would be 2).
+    :return: Table of loci in sections where no breaks were found in the break search round. Schema:
+        ----------------------------------------
+        Row fields:
+            'locus': locus<GRCh37>
+            'section': str
+            'section_obs': int64
+            'section_exp': float64
+            'section_oe': float64
+        ----------------------------------------
+        Key: ['locus', 'section']
+        ----------------------------------------
+    """
+    single_no_break_path = single_search_round_ht_path(
+        is_rescue=is_rescue,
+        search_num=search_num,
+        is_break_found=False,
+        is_breakpoint_only=False,
     )
-    return ht.annotate(
-        section_chisq=calculate_section_chisq(ht.section_obs, ht.section_exp)
+    if file_exists(single_no_break_path):
+        raise DataException(
+            f"No table found at {single_no_break_path}. Please double-check!"
+        )
+    ht = hl.read_table(single_no_break_path)
+    ht = ht.select(*FINAL_ANNOTATIONS)
+    simul_results_path = simul_search_round_bucket_path(
+        is_rescue=is_rescue,
+        search_num=search_num,
+        bucket_type="final_results",
     )
+    simul_break_path = f"{simul_results_path}/merged.ht"
+    if file_exists(simul_break_path):
+        simul_ht = hl.read_table(simul_break_path)
+        simul_sections = simul_ht.aggregate(hl.agg.collect_as_set(simul_ht.section))
+        ht = ht.filter(~hl.literal(simul_sections).contains(ht.section))
```

Comment:
```suggestion
    if file_exists(simul_break_path):
        simul_ht = hl.read_table(simul_break_path)
        simul_sections = simul_ht.aggregate(hl.agg.collect_as_set(simul_ht.section))
        ht = ht.filter(~hl.literal(simul_sections).contains(ht.section))
    else:
        logger.warning("Simul breaks results HT did not exist. Please double check that this was expected!")
```
(hopefully I didn't break the formatting)
this table won't always exist, but I thought a warning might be helpful

Reply:
Just read this over again and totally agree!